app/models/text.rb

```diff
@@ -7,4 +7,23 @@ class Text < ApplicationRecord
   def read_by?(user)
     reads.find_by(user_id: user.id).present?
   end
-end
+
+  # ジャンル別のデータ数を配列で取得
+  def self.total_count
+    self.group(:genre).count
+  end
+
+  # 読破済みの数を配列で取得
+  def self.read_count(user)
+    user.read_texts.group(:genre).count
+  end
+
+  # 読破済み進捗のパーセンテージ
+  def self.percentage(read_count, total_count)
+    if total_count.zero?
+      return 0
```

Comment:
`return` はなくてもよいですね。

```suggestion
      0
```